src/rt-app.c

```diff
@@ -1031,6 +1031,23 @@ static void set_thread_priority(thread_data_t *data, sched_data_t *sched_data)
 		sa_params.sched_flags = SCHED_FLAG_KEEP_ALL;
 		tid = gettid();
 
+		/*
+		 * Currently we cannot use the syscall to set a clamp
+		 * value without specifing a valid policy and priotiry.
+		 * Read the current values and use them for the following
+		 * syscall.
+		 * TODO: fix __sched_setscheduler() to get rid of this!
+		 */
+		ret = sched_getattr(tid, &sa_params,
+				    sizeof(struct sched_attr), flags);
+		if (ret != 0) {
+			log_critical("[%d] sched_getattr returned %d",
+					data->ind, ret);
+			errno = ret;
+			perror("sched_getattr");
+			exit(EXIT_FAILURE);
+		}
+
```

Comment:
I remember discussing this with Qais.
We already know which policy is currently set so we don't need the syscall which is costly

Reply:
The problem isn't the policy but the priority. In RT case, `sched_data->prio` is `sched_priority`, it is working. In the case of CFS though, `sched_priority` has no meaning and it should be set to a default value (while `sched_data->prio` is the nice value (which can be negative... while `sched_priority` is unsigned). We could hardcode this value here, but there's no guarantee that it won't change in the kernel. That's why I believe it is better to read the `sched_priority` value, instead of writing it.

In any case, `sa_params.sched_priority = sched_data->prio` is wrong. See `__sched_setscheduler`

```        /*
         * Valid priorities for SCHED_FIFO and SCHED_RR are
         * 1..MAX_USER_RT_PRIO-1, valid priority for SCHED_NORMAL,
         * SCHED_BATCH and SCHED_IDLE is 0.
         */
```

Reply:
so set sa_params.sched_priority to 0 for CFS policy